pkg/controllers/utils/common.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+// UpdateFailoverStatus adds a failoverHistoryItem to the failoverHistory field in the ResourceBinding.
+func UpdateFailoverStatus(client client.Client, binding *workv1alpha2.ResourceBinding, failoverType workv1alpha2.FailoverReason) (err error) {
+	klog.V(4).Infof("Updating failover status for ResourceBinding(%s/%s)", binding.Name, binding.Namespace)
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
+		_, err = helper.UpdateStatus(context.Background(), client, binding, func() error {
+			failoverHistoryItem := workv1alpha2.FailoverHistoryItem{
+				StartTime:              metav1.Time{Time: time.Now()},
+				Reason:                 failoverType,
+				ClustersBeforeFailover: binding.Spec.Clusters,
```

Comment:
The assignment here seems to be inconsistent with the [comment](https://github.com/karmada-io/karmada/pull/5251/files#r1818329655), where the ClustersBeforeFailover assignment should be all clusters that were last scheduled.

Reply:
You're totally right, this should be the clusters that are used in `evictBinding`.